test/functional/page_objects/visual_builder_page.js

```diff
@@ -69,23 +82,114 @@ export function VisualBuilderPageProvider({ getService, getPageObjects }) {
       }
       await input.pressKeys(browser.keys.NULL); // Release modifier keys
       await input.pressKeys(browser.keys.BACKSPACE); // Delete all content
-      await input.type(markdown);
       await PageObjects.visualize.waitForRenderingCount(prevRenderingCount + 1);
     }
-
     async getMarkdownText() {
       const el = await find.byCssSelector('.tvbEditorVisualization');
-      return await el.getVisibleText();
+      const text = await el.getVisibleText();
+      return text;
+    }
+
+    /**
+     *
+     * getting all markdown variables list which located on `table` section
+     *
+     * **Note**: if `table` not have variables, use `getMarkdownTableNoVariables` method instead
+     * @see {getMarkdownTableNoVariables}
+     * @returns {Promise<Array<{key:string, value:string, selector:any}>>}
+     * @memberof VisualBuilderPage
+     */
+    async getMarkdownTableVariables() {
+      const testTableVariables = await testSubjects.find('tsvbMarkdownVariablesTable');
+      const variablesSelector = 'tbody tr';
+      const exists = await find.existsByDisplayedByCssSelector(variablesSelector);
+      if (!exists) {
+        log.debug('variable list is empty');
+        return [];
+      }
+      const variables = await testTableVariables.findAllByCssSelector(variablesSelector);
+
+      const variablesKeyValueSelectorMap = await Promise.all(variables.map(async variable => {
+        const subVars = await variable.findAllByCssSelector('td');
+        const selector = subVars[0];
+        const key = await selector.getVisibleText();
+        const value = await subVars[1].getVisibleText();
+        return { key, value, selector };
+      }));
+      log.debug(`markdown variables table is: ${variablesKeyValueSelectorMap}`);
```

Comment:
I've just noticed that this map is logged as `[object Object],[object Object],[object Object],[object Object],[object Object]`
I don't think that it's valuable.